### PR TITLE
Launcher for automatic perf tests

### PR DESCRIPTION
**Pull request by Paolob69**
Python script that manages automatic performance tests using command line parameters and report the results on csv file

### CODE REVIEW

File: tests/perf/perf.py [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+        self.get_result(index, name, qps_value, time_value)
+
+
+    def get_result(self, test_number, name, qps_value, time_value):
+        """ Processes the report file generated by vegeta and reads latency data
+        """
+        test_report_filename = "./getLogs_"+str(qps_value)+"qps_"+str(time_value)+"s_"+name+"_perf.hrd"
+        file = open(test_report_filename)
+        try:
+            thrds = os.popen("ps -efL | grep tg | wc -l").read().replace('\n', ' ')
```
Comment (canepat) [inline tests/perf/perf.py:144]: `thrds` -> `threads`

File: tests/perf/perf.py [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+        self.writer.writerow(["", "", "", "", "", "", "", "", "", "", "", "", "PC", model[1]])
+        self.writer.writerow(["", "", "", "", "", "", "", "", "", "", "", "", "DaemonRunOnCore", self.config_class.daemon_on_core])
+        self.writer.writerow(["", "", "", "", "", "", "", "", "", "", "", "", "VegetaFile", self.config_class.pattern_tar_file])
+        self.writer.writerow(["", "", "", "", "", "", "", "", "", "", "", "", "VegetaChecksum", checksum[0]])
+        self.writer.writerow(["", "", "", "", "", "", "", "", "", "", "", "", "GccVers", gcc_vers[0]])
+        self.writer.writerow(["", "", "", "", "", "", "", "", "", "", "", "", "SilkVersion", "TBD"])
+        self.writer.writerow(["", "", "", "", "", "", "", "", "", "", "", "", "RpcDaemon", "TBD"])
+        self.writer.writerow([])
+        self.writer.writerow([])
+        self.writer.writerow(["Daemon", "TestNo", "TG-Thrs", "Qps", "Time", "Min", "Mean", "50", "90", "95", "99", "Max"])
```
Comment (canepat) [inline tests/perf/perf.py:195]: `"TG-Thrs"` -> `"TG-Threads"`

File: tests/perf/perf.py [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+        self.csv_file.flush()
+
+
+    def write_test_report(self, daemon, test_number, thrds, qps_value, time_value, min_latency, mean, fifty, ninty, nintyfive, nintynine, max_latency):
```
Comment (canepat) [inline tests/perf/perf.py:199]: `thrds` -> `threads`

File: tests/perf/perf.py [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+# main
+#
+# usage: python perf.py <patternTarFile> [daemonOnCore - or 0-1 or 0-2 ...] [gethNode localhost or IP]
+#                       [geth_homedir i.e ../../../turbo-geth/] [test rep i.e 10] [test seq i.e 200+30-400+10]
+#
+def main(argv):
+    """ Executes tests on selected user configuration
+    """
+
+    config_class = Config(argv)
```
Comment (canepat) [inline tests/perf/perf.py:223]: `config_class` -> `config`

**Review by canepat — CHANGES_REQUESTED** [review]

File: tests/perf/perf.py [hunk 5]
```diff
@@ -0,0 +1,263 @@
+""" This script is responsible using vegeta to make a list of performance tests (configured via command line) and saves its result on csv file
+"""
+
+import os
+import csv
+import sys
+from datetime import datetime
+
+DEFAULT_TEST_SEQUENCE = "50+30-200+30-200+60-400+30"
```
[the remark this answers is not in the document]
Comment (Paolob69) [reply tests/perf/perf.py:9]: ok

File: tests/perf/perf.py [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+DEFAULT_REPETITION_VALUE = 10
```
[the remark this answers is not in the document]
Comment (Paolob69) [reply tests/perf/perf.py:10]: ok

File: tests/perf/perf.py [hunk 7]
```diff
@@ -0,0 +1,263 @@
...
+DEFAULT_PATTERN_TAR_FILE = ""
```
[the remark this answers is not in the document]
Comment (Paolob69) [reply tests/perf/perf.py:11]: ok

File: tests/perf/perf.py [hunk 8]
```diff
@@ -0,0 +1,263 @@
...
+
+
+class Config:
+    """ This class manage configuration params
+    """
+    def __init__(self, argv):
+        """ Processes the command line contained in argv
+        """
+        self.test_sequence = DEFAULT_TEST_SEQUENCE
+        self.rep_tests = DEFAULT_REPETITION_VALUE
```
[the remark this answers is not in the document]
Comment (Paolob69) [reply tests/perf/perf.py:24]: ok

File: tests/perf/perf.py [hunk 9]
```diff
@@ -0,0 +1,263 @@
...
+        self.pattern_tar_file = DEFAULT_PATTERN_TAR_FILE
```
[the remark this answers is not in the document]
Comment (Paolob69) [reply tests/perf/perf.py:25]: ok

File: tests/perf/perf.py [hunk 10]
```diff
@@ -0,0 +1,263 @@
...
+        elif len(argv) != 2:
+            print("usage: "+argv[0]+ " <patternTarFile> [daemonOnCore - or 0-1 or 0-2 ...]")
+            print("               [gethNode localhost or IP] [geth_homedir i.e ../../../turbo-geth/] [test rep i.e 10] [test seq i.e 200+30-400+10]")
+            sys.exit(-1)
+
+
+class PerfTest:
+    """ This class manage performance test
+    """
+    def __init__(self, test_report, config_class):
```
[the remark this answers is not in the document]
Comment (Paolob69) [reply tests/perf/perf.py:66]: ok

File: tests/perf/perf.py (hunk 1 above)
Comment (Paolob69) [reply tests/perf/perf.py:144]: ok

File: tests/perf/perf.py (hunk 2 above)
Comment (Paolob69) [reply tests/perf/perf.py:195]: ok

File: tests/perf/perf.py (hunk 3 above)
Comment (Paolob69) [reply tests/perf/perf.py:199]: ok

File: tests/perf/perf.py (hunk 4 above)
Comment (Paolob69) [reply tests/perf/perf.py:223]: ok

File: tests/perf/perf.py [hunk 11]
```diff
@@ -0,0 +1,263 @@
...
+            self.geth_homedir = argv[4]
+            self.rep_tests = int(argv[5])
+        elif len(argv) == 7:
+            self.pattern_tar_file = argv[1]
+            self.daemon_on_core = argv[2]
+            self.geth_node_addr = argv[3]
+            self.geth_homedir = argv[4]
+            self.rep_tests = int(argv[5])
+            self.test_sequence = argv[6]
+        elif len(argv) != 2:
```
[the remark this answers is not in the document]
Comment (Paolob69) [reply tests/perf/perf.py:57]: ok

**Review by canepat — APPROVED** [review]

**Pull request merged**